AndroidSDK/src/com/leanplum/LeanplumNotificationHelper.java

```diff
@@ -249,4 +258,64 @@ protected RemoteViews getStandardView(int layoutId) {
     notificationBuilder.setContentIntent(contentIntent);
     return notificationBuilder;
   }
+
+  /**
+   * Checks a possibility to create icon drawable from current app icon.
+   *
+   * @param context Current application context.
+   * @return boolean True if it is possible to create a drawable from current app icon.
+   */
+  private static boolean canCreateIconDrawable(Context context) {
+    try {
+      // Try to create icon drawable.
+      Drawable drawable = AdaptiveIconDrawable.createFromStream(
+          context.getResources().openRawResource(context.getApplicationInfo().icon),
+          "applicationInfo.icon");
+      // If there was no crash, we still need to check for null.
+      if (drawable != null) {
+        return true;
+      }
+    } catch (Throwable ignored) {
+    }
+    return false;
+  }
+
+  /**
+   * Validation of Application icon for small icon on push notification.
+   *
+   * @param context Current application context.
+   * @return boolean True if application icon can be used for small icon on push notification.
+   */
+  static boolean isApplicationIconValid(Context context) {
+    if (context == null) {
+      return false;
+    }
+
+    //TODO Potentially there should be checked for Build.VERSION.SDK_INT != 26, but we need to
+    //TODO confirm that adaptive icon works well on 27, before to change it.
+    // Checks current version of device. If current version less than 26, we can skip next step.
```

Comment:
move this line up as the first line of comment.
What is the next step? It's prob. enough to describe what this method is doing only.